### DISCUSSION TITLE
Does Selenide support automating CEF? and how we can do it?

### DISCUSSION THREAD

**Discussion by ovclara** [selenide/selenide#2340] [Q&A]
Hello,

I am new to Selenide (and selenium too) .
I  liked selenide due to its easy syntax and great abilities, which inspired me to explore it to create a POC for my project.

It was easy to get started with automating web application but not sure how I do it for CEF, below is the work flow I am trying to automate.
1. Launch the app in Web do operations on web elements.
2. The from Web > Launch a Desktop application 
3. The Desktop application has a CEF component in it
4. Need to shift the focus from Web to the CEF and automate CEF operations.

Automating Desktop is done thru AutoITX4Java and I am looking how selenide can handle the case:
--> switching between browser > CEF and vise versa
--> Switching between browser > Desktop application > CEF and vise versa

Please guide me if it is possible.

**Comment by asolntsev** [comment]
@ovclara Em... Does CEF mean "Connecting Europe Facility" or "Center of Excellence in Finance"? Or maybe "Cisco Express Forwarding"?

I have no experience with CEF, but I see there are some answers in google: https://stackoverflow.com/questions/57818290/how-to-write-selenium-tests-for-cef-applications

**Comment by ovclara** [comment]
Ohh.. Sorry , should have been precise. 
 Its Chromium Embedded Framework (CEF).
You pointed to the right CEF stackoverflow page.

So, it will be the same kind of settings required for Selenide as well ?
I will try to see if I can get it set-up for my task.
I guess if we can have an example for selenide with Chromium Embedded Framework (CEF) , this will be very helpful.

**Comment by asolntsev** [comment]
@ovclara Yes, you would need a similar setup in Selenide. Something like

```java
import com.codeborne.selenide.Configuration;
import static com.codeborne.selenide.Selenide.open;

ChromeOptions options = new ChromeOptions();
// Path to the CEF executable.
options.setBinary("c:/temp/cef_binary_3.2171.1979_windows32_client/Release/cefclient.exe");
Configuration.browserCapabilities = options; 

open("https://whatever.com");
```
